resolver/caching_resolver.go

```diff
@@ -212,21 +212,41 @@ func (r *CachingResolver) trackQueryDomainNameCount(domain, cacheKey string, log
 	}
 }
 
+// removes EDNS OPT records from message
+func removeEdns0Extra(msg *dns.Msg) {
+	if len(msg.Extra) > 0 {
+		extra := make([]dns.RR, 0)
```

Comment:
We can pre-alloc the array:
```suggestion
		extra := make([]dns.RR, 0, len(msg.Extra))
```

Reply:
Thanks, this solution is much better